### PR TITLE
[release-1.5] 🌱 Bump corefile-migration library to v1.0.21

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
Manual cherry-pick of #9307

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/book/src/reference/versions.md [hunk 1]
```diff
@@ -144,6 +144,9 @@ The Kubeadm Control Plane requires the Kubeadm Bootstrap Provider.
 | v1.3 (v1beta1)       | v1.10.0                         |
 | >= v1.3.4 (v1beta1)  | v1.10.1                         |
 | v1.4 (v1beta1)       | v1.10.1                         |
+| >= v1.4.6 (v1beta1)  | v1.11.1                         |
+| v1.5 (v1beta1)       | v1.10.1                         |
```
Comment (furkatgofurov7) [inline docs/book/src/reference/versions.md:148]: I overlooked it in other PR, is this correct?

**Review by furkatgofurov7 — COMMENTED** [review]
/area dependency

File: docs/book/src/reference/versions.md (hunk 1 above)
Comment (sbueringer) [reply docs/book/src/reference/versions.md:148]: Yes it is correct

Comment (sbueringer) [reply docs/book/src/reference/versions.md:148]: Oure CoreDNS support in this table depends on the migration library we use

Comment (sbueringer) [reply docs/book/src/reference/versions.md:148]: You're welcome. No problem :) Thx for asking

Comment (sbueringer) [reply docs/book/src/reference/versions.md:148]: Just assuming PR authors know what they're doing doesn't help much on PRs :joy:

**Review by killianmuldoon — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**